src/rust/engine/process_execution/src/lib.rs

```diff
@@ -341,13 +344,54 @@ pub struct ProcessMetadata {
 ///
 /// The result of running a process.
 ///
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Derivative, Clone, Debug, Eq)]
+#[derivative(PartialEq, Hash)]
 pub struct FallibleProcessResultWithPlatform {
   pub stdout_digest: Digest,
   pub stderr_digest: Digest,
   pub exit_code: i32,
-  pub platform: Platform,
   pub output_directory: hashing::Digest,
+  pub platform: Platform,
+  #[derivative(PartialEq = "ignore", Hash = "ignore")]
+  pub metadata: ProcessResultMetadata,
+}
+
+/// Metadata for a ProcessResult corresponding to the REAPI `ExecutedActionMetadata` proto.
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct ProcessResultMetadata {
+  pub execution_time: Option<TimeSpan>,
```

Comment:
Since this type doesn't actually have a use for the start time, this could maybe just be a Duration, and then the only case in which we'd need to be concerned with a start time woul be in `impl Into<ExecutedActionMetadata> for ProcessResultMetadata`... where we could use a static/fixed start time, afaict.

You still wouldn't want it in the `Hash`/`Eq` of `ProcessResult`... just a bit simpler to construct/work with.

Reply:
> where we could use a static/fixed start time, afaict.

I was thinking about this too. I do think it'd work. But I went with `TimeSpan` once I discovered it exists because that way we avoid dropping any information. I don't think we need that start time now, but maybe relevant in the future. In this way, we're more of a 1-1 mapping between `ExecutedActionMetadata` and `ProcessResultMetadata`.

> just a bit simpler to construct/work with.

I'm not as concerned about this as I was at first because it's encapsulated into the `Into` and `From` traits.

Reply:
> I'm not as concerned about this as I was at first because it's encapsulated into the Into and From traits.

Only because you're currently constructing a "default" `ProcessResultMetadata`: when you start actually constructing them for local execution you'll need to create a `TimeSpan` rather than just a `Duration`.

(... which reminds me that we probably shouldn't implement `Default` there: `ProcessResultMetadata` should probably have a required `Duration`.)

Reply:
> when you start actually constructing them for local execution you'll need to create a TimeSpan rather than just a Duration.

True, will see how painful this.

> (ProcessResultMetadata should probably have a required Duration.)

I don't think we can do this because we are not guaranteed remote execution has set the relevant fields in `ExecutedActionMetadata`, right?

Reply:
> I don't think we can do this because we are not guaranteed remote execution has set the relevant fields in ExecutedActionMetadata, right?

Oh. Maybe. I wonder if we'd want to just estimate in that case with a local calculation.

Reply:
K yeah you're right that creating a `TimeSpan` is tricky for local processes - the Rust docs warn `SystemTime` is not monotonic and not as safe as `Instant` for calculating how long something took.

So looks like fuzzing the start time is the way to go.